diff --git a/Source/GlobalEditor.cpp b/Source/GlobalEditor.cpp
index 2efeca0..bc7b5c5 100644
--- a/Source/GlobalEditor.cpp
+++ b/Source/GlobalEditor.cpp
@@ -78,7 +78,7 @@ public:
     std::unique_ptr<juce::HyperlinkButton> dexed; // changed to std::unique_ptr from juce::ScopedPointer
     std::unique_ptr<juce::HyperlinkButton> surge; // changed to std::unique_ptr from juce::ScopedPointer
 
-    AboutBox() : DialogWindow("About", Colour(0xFF000000), true),
+    AboutBox() : DialogWindow("About", Colour(0xFF000000), true, false),
         dexed(std::make_unique<juce::HyperlinkButton>("https://asb2m10.github.io/dexed/", URL("https://asb2m10.github.io/dexed/"))),
         surge(std::make_unique<juce::HyperlinkButton>("https://surge-synthesizer.github.io/", URL("https://surge-synthesizer.github.io/")))
     {
@@ -702,8 +702,13 @@ void GlobalEditor::buttonClicked (juce::Button* buttonThatWasClicked)
     {
         //[UserButtonCode_aboutButton] -- add your button handler code here..
         aboutBox = std::make_unique<AboutBox>();
-        getParentComponent()->addAndMakeVisible(aboutBox.get());
+        auto* parent = getTopLevelComponent();
+        if (parent == nullptr)
+            parent = this;
+
+        parent->addAndMakeVisible(aboutBox.get());
         aboutBox->centreWithSize(aboutBox->getWidth(), aboutBox->getHeight());
+        aboutBox->toFront(true);
         aboutBox->enterModalState(true,{});
         //[/UserButtonCode_aboutButton]
     }
diff --git a/Source/DXComponents.cpp b/Source/DXComponents.cpp
index d831a5f..2f07236 100644
--- a/Source/DXComponents.cpp
+++ b/Source/DXComponents.cpp
@@ -374,6 +374,7 @@ static void comboBoxPopupMenuFinishedCallback (int result, ComboBoxImage* combo)
 
 void ComboBoxImage::showPopup() {
     popup.showMenuAsync (PopupMenu::Options().withTargetComponent (this)
+                         .withParentComponent(getTopLevelComponent())
                          .withItemThatMustBeVisible(getSelectedId())
                          .withMinimumWidth(getWidth())
                          .withMaximumNumColumns(1)
diff --git a/Source/DXLookNFeel.h b/Source/DXLookNFeel.h
index a9ee3af..6a050d2 100644
--- a/Source/DXLookNFeel.h
+++ b/Source/DXLookNFeel.h
@@ -52,6 +52,7 @@ public:
     //virtual Font getTextButtonFont(TextButton&, int buttonHeight) override;
     virtual Typeface::Ptr getTypefaceForFont(const Font &) override;
     virtual void positionComboBoxText (ComboBox& box, Label& label) override;
+    virtual PopupMenu::Options getOptionsForComboBoxPopupMenu (ComboBox& box, Label& label) override;
 
     static Colour fillColour;
     static Colour lightBackground;
diff --git a/Source/DXLookNFeel.cpp b/Source/DXLookNFeel.cpp
index e238e08..3da2772 100644
--- a/Source/DXLookNFeel.cpp
+++ b/Source/DXLookNFeel.cpp
@@ -355,6 +355,15 @@ void DXLookNFeel::positionComboBoxText(ComboBox& box, Label& label) {
     LookAndFeel_V4::positionComboBoxText(box, label);
 }
 
+PopupMenu::Options DXLookNFeel::getOptionsForComboBoxPopupMenu(ComboBox& box, Label& label) {
+    auto options = LookAndFeel_V4::getOptionsForComboBoxPopupMenu(box, label);
+
+    if (auto* parent = box.getTopLevelComponent())
+        options = options.withParentComponent(parent);
+
+    return options;
+}
+
 Colour DXLookNFeel::fillColour = Colour(77,159,151);
 Colour DXLookNFeel::lightBackground = Colour(78,72,63);
 Colour DXLookNFeel::background = Colour(60,50,47);
