diff --git a/CMakeLists.txt b/CMakeLists.txt
index f81961f..fe28c05 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -19,7 +19,7 @@ if (${CMAKE_SYSTEM_NAME} STREQUAL "iOS")
 #    set(CMAKE_XCODE_ATTRIBUTE_CODE_SIGN_IDENTITY "iPhone Developer" CACHE STRING "" FORCE)
 #    set(CMAKE_XCODE_ATTRIBUTE_DEVELOPMENT_TEAM "XXXXXXXXXX" CACHE STRING "" FORCE)
 else()
-    set(CMAKE_OSX_DEPLOYMENT_TARGET "10.9" CACHE STRING "Minimum OS X deployment version" FORCE)
+    set(CMAKE_OSX_DEPLOYMENT_TARGET "11.0" CACHE STRING "Minimum OS X deployment version" FORCE)
 endif()
 
 if (CMAKE_BUILD_TYPE STREQUAL "Release")
diff --git a/Source/CMakeLists.txt b/Source/CMakeLists.txt
index 3f06b02..45bd934 100644
--- a/Source/CMakeLists.txt
+++ b/Source/CMakeLists.txt
@@ -114,3 +114,54 @@ PUBLIC
     juce::juce_recommended_config_flags
     juce::juce_recommended_lto_flags
 )
+
+# begin JUCEAAP specific -->
+
+include_directories(
+		"${AAP_DIR}/include"
+		"${JUCE_DIR}/modules"
+)
+
+juce_add_modules(${AAP_JUCE_DIR}/aap-modules/aap_audio_processors)
+target_link_libraries(${BaseTargetName}
+		PRIVATE
+		aap_audio_processors
+		)
+
+add_compile_definitions(
+		[[JUCEAAP_USE_UNIQUE_ID=1]]
+		[[JUCEAAP_HOSTED_PARAMETER=1]]
+		[[JUCEAAP_HAVE_AUDIO_PLAYHEAD_NEW_POSITION_INFO=1]]
+)
+
+if (ANDROID)
+add_library("cpufeatures" STATIC "${ANDROID_NDK}/sources/android/cpufeatures/cpu-features.c")
+set_source_files_properties("${ANDROID_NDK}/sources/android/cpufeatures/cpu-features.c" PROPERTIES COMPILE_FLAGS "-Wno-sign-conversion -Wno-gnu-statement-expression")
+
+# dependencies
+find_library(log "log")
+find_library(android "android")
+find_library(glesv2 "GLESv2")
+find_library(egl "EGL")
+set(cpufeatures_lib "cpufeatures")
+set(oboe_lib "oboe")
+set(CMAKE_CXX_FLAGS_RELEASE "-O3")
+
+add_compile_definitions(
+		[[JUCE_ANDROID=1]]
+		[[JUCE_DONT_AUTO_OPEN_MIDI_DEVICES_ON_MOBILE=1]]
+		[[JUCE_PUSH_NOTIFICATIONS_ACTIVITY="com/rmsl/juce/JuceActivity"]]
+		[[JUCE_PUSH_NOTIFICATIONS=1]]
+		[[JUCE_USE_ANDROID_OBOE_STABILIZED_CALLBACK=1]]
+)
+
+target_link_libraries(${BaseTargetName}
+    PRIVATE
+    ${log}
+    ${android}
+    ${glesv2}
+    ${egl}
+    ${cpufeatures_lib}
+    )
+endif (ANDROID)
+# <-- end JUCEAAP specific
diff --git a/Source/CartManager.cpp b/Source/CartManager.cpp
index 779026e..5417782 100644
--- a/Source/CartManager.cpp
+++ b/Source/CartManager.cpp
@@ -265,10 +265,10 @@ void CartManager::buttonClicked(juce::Button *buttonThatWasClicked) {
     if ( buttonThatWasClicked == loadButton.get() ) {
         FileChooser fc ("Import original DX sysex...", File::getSpecialLocation(File::SpecialLocationType::userDocumentsDirectory), "*.syx;*.SYX;*.*", 1);
 
-        if ( fc.browseForFileToOpen()) {
+        fc.launchAsync(juce::FileBrowserComponent::openMode, [&](const juce::FileChooser &_fc) {
             mainWindow->loadCart(fc.getResult());
             updateCartFilename();
-        }
+        });
         return;
     }
 
@@ -322,7 +322,8 @@ void CartManager::fileClicked(const File& file, const MouseEvent& e) {
         menu.addSeparator();
         menu.addItem(1020, "Refresh");
 
-        switch(menu.show()) {
+        menu.showMenuAsync(PopupMenu::Options{}, [=](int result) {
+        switch(result) {
         case 1000:
             file.revealToUser();
             break;
@@ -332,7 +333,7 @@ void CartManager::fileClicked(const File& file, const MouseEvent& e) {
         case 1020:
             cartBrowserList->refresh();
             break;
-        }
+        }});
         return;
     }
 }
@@ -382,7 +383,8 @@ void CartManager::programRightClicked(ProgramListBox *source, int pos) {
     if ( source == activeCart.get() )
         menu.addItem(1010, "Send current sysex cartridge to DX7");
 
-    switch(menu.show())  {
+    menu.showMenuAsync(PopupMenu::Options{}, [&](int result) {
+    switch(result)  {
         case 1000:
             uint8_t unpackPgm[161];
 
@@ -403,7 +405,7 @@ void CartManager::programRightClicked(ProgramListBox *source, int pos) {
         case 1010:
             mainWindow->processor->sendCurrentSysexCartridge();
             break;
-    }
+    }});
 }
 
 void CartManager::programDragged(ProgramListBox *destListBox, int dest, char *packedPgm) {
diff --git a/Source/GlobalEditor.cpp b/Source/GlobalEditor.cpp
index 50aa618..fffa266 100644
--- a/Source/GlobalEditor.cpp
+++ b/Source/GlobalEditor.cpp
@@ -794,7 +794,7 @@ void GlobalEditor::mouseDown(const MouseEvent &e) {
         PopupMenu popup;
         popup.addItem(1, "Send current program to DX7");
 
-        auto p = popup.show();
+        popup.showMenuAsync(PopupMenu::Options{}, [=](int p) {
         switch( p )
         {
         case 1:
@@ -802,7 +802,7 @@ void GlobalEditor::mouseDown(const MouseEvent &e) {
            break;
         default:
             break;
-        }
+        }});
     }
 }
 //[/MiscUserCode]
diff --git a/Source/OperatorEditor.cpp b/Source/OperatorEditor.cpp
index 426c016..939b795 100644
--- a/Source/OperatorEditor.cpp
+++ b/Source/OperatorEditor.cpp
@@ -612,7 +612,8 @@ void OperatorEditor::mouseDown(const MouseEvent &event) {
             popup.addItem(5, "Reset plugin UI scaling factor");
         }
 
-        switch(popup.show()) {
+        popup.showMenuAsync(PopupMenu::Options{},[&](int result) {
+        switch(result) {
             case 1:
                 processor->copyToClipboard(internalOp);
             break;
@@ -635,7 +636,7 @@ void OperatorEditor::mouseDown(const MouseEvent &event) {
                     editor->resetZoomFactor();
                 }
             break;
-        }
+        }});
 
     }
 }
diff --git a/Source/PluginEditor.cpp b/Source/PluginEditor.cpp
index 44ce786..4a322a6 100644
--- a/Source/PluginEditor.cpp
+++ b/Source/PluginEditor.cpp
@@ -142,10 +142,10 @@ void DexedAudioProcessorEditor::loadCart(File file) {
     }
 
     if ( rc != 0 ) {
-        rc = AlertWindow::showOkCancelBox(AlertWindow::QuestionIcon, "Unable to find DX7 sysex cartridge in file",
+        /*rc = AlertWindow::showOkCancelBox(AlertWindow::QuestionIcon, "Unable to find DX7 sysex cartridge in file",
                                           "This sysex file is not for the DX7 or it is corrupted. "
                                           "Do you still want to load this file as random data ?");
-        if ( rc == 0 )
+        if ( rc == 0 )*/
             return;
     }
 
@@ -162,13 +162,13 @@ void DexedAudioProcessorEditor::saveCart() {
     File startFileName = processor->activeFileCartridge.exists() ? processor->activeFileCartridge : processor->dexedCartDir;
 
     FileChooser fc ("Export DX sysex...", processor->dexedCartDir, "*.syx;*.SYX", 1);
-    if ( fc.browseForFileToSave(true) ) {
+    fc.launchAsync(FileBrowserComponent::saveMode, [=](const FileChooser& fc) {
         if ( ! processor->currentCart.saveVoice(fc.getResults().getReference(0)) ) {
             AlertWindow::showMessageBoxAsync (AlertWindow::WarningIcon,
                                               "Error",
                                               "Unable to write: " + fc.getResults().getReference(0).getFullPathName());
         }
-    }
+    });
 }
 
 void DexedAudioProcessorEditor::tuningShow() {
@@ -322,6 +322,9 @@ void DexedAudioProcessorEditor::rebuildProgramCombobox() {
 }
 
 void DexedAudioProcessorEditor::storeProgram() {
+#if ANDROID
+    AlertWindow::showMessageBoxAsync(AlertWindow::WarningIcon, "Not Supported", "We don't support storing proglrams on Android build yet.");
+#else
     String currentName = Cartridge::normalizePgmName((const char *) processor->data+145);
     Cartridge destSysex = processor->currentCart;
     File *externalFile = NULL;
@@ -420,6 +423,7 @@ void DexedAudioProcessorEditor::storeProgram() {
     if ( externalFile != NULL )
         delete externalFile;
     cartManager.resetActiveSysex();
+#endif
 }
 
 class MidiCCListener: public AlertWindow, Value::Listener {
@@ -449,8 +453,12 @@ public :
 };
 
 void DexedAudioProcessorEditor::discoverMidiCC(Ctrl *ctrl) {
+#if ANDROID
+    AlertWindow::showMessageBoxAsync(AlertWindow::WarningIcon, "Not Supported", "We don't support storing proglrams on Android build yet.");
+#else
     MidiCCListener ccListener(this, ctrl);
     ccListener.runModalLoop();
+#endif
 }
 
 float DexedAudioProcessorEditor::getLargestScaleFactor() {
@@ -509,14 +517,14 @@ void DexedAudioProcessorEditor::filesDropped (const StringArray &files, int x, i
         if (fn.endsWithIgnoreCase(".scl"))
         {
             if (filesize == 0) {
-                AlertWindow::showMessageBox(
+                AlertWindow::showMessageBoxAsync(
                     AlertWindow::WarningIcon,
                     "File size error!",
                     "File \'" + fn.toStdString() + "\' is empty."
                 );
             }
             else if (filesize > MAX_SCL_KBM_FILE_SIZE) {
-                AlertWindow::showMessageBox(
+                AlertWindow::showMessageBoxAsync(
                     AlertWindow::WarningIcon,
                     "File size error!",
                     "File \'" + fn.toStdString() + "\' has " + std::to_string(filesize) + " bytes, exceeding the maximum limit ("+std::to_string(MAX_SCL_KBM_FILE_SIZE)+")."
@@ -529,14 +537,14 @@ void DexedAudioProcessorEditor::filesDropped (const StringArray &files, int x, i
         if (fn.endsWithIgnoreCase(".kbm"))
         {
             if (filesize == 0) {
-                AlertWindow::showMessageBox(
+                AlertWindow::showMessageBoxAsync(
                     AlertWindow::WarningIcon,
                     "File size error!",
                     "File \'" + fn.toStdString() + "\' is empty."
                 );
             }
             else if (filesize > MAX_SCL_KBM_FILE_SIZE) {
-                AlertWindow::showMessageBox(
+                AlertWindow::showMessageBoxAsync(
                     AlertWindow::WarningIcon,
                     "File size error!",
                     "File \'" + fn.toStdString() + "\' has " + std::to_string(filesize) + " bytes, exceeding the maximum limit (" + std::to_string(MAX_SCL_KBM_FILE_SIZE) + ")."
@@ -548,20 +556,20 @@ void DexedAudioProcessorEditor::filesDropped (const StringArray &files, int x, i
         }
     }
     catch (const std::ios_base::failure& ex) {
-        AlertWindow::showMessageBox(
+        AlertWindow::showMessageBoxAsync(
             AlertWindow::WarningIcon, 
             "I/O error!", 
             "Related to file \'" + fn.toStdString() + "\', an exception (std::ios_base::failure) occured: " + ex.what()
         );
     }
     catch (std::bad_alloc& ex) {
-        AlertWindow::showMessageBox(
+        AlertWindow::showMessageBoxAsync(
             AlertWindow::WarningIcon, 
             "I/O error!", 
             "Related to file \'" + fn.toStdString() + "\', an exception (std::bad_alloc) occured: " + ex.what());
     }
     catch (...) {
-        AlertWindow::showMessageBox(
+        AlertWindow::showMessageBoxAsync(
             AlertWindow::WarningIcon, 
             "I/O error!", 
             "Related to file \'"+fn.toStdString()+"\', an unknown exception occured.");
diff --git a/Source/PluginParam.cpp b/Source/PluginParam.cpp
index c824bb7..f9a56a6 100644
--- a/Source/PluginParam.cpp
+++ b/Source/PluginParam.cpp
@@ -142,16 +142,17 @@ void Ctrl::mouseDown(const juce::MouseEvent &event) {
             popup.addItem(5, "Reset plugin UI scaling factor");
         }
 
-        switch(popup.show()) {
+        popup.showMenuAsync(PopupMenu::Options{}, [&](int result) {
+        switch(result) {
             case 1:
                 parent->mappedMidiCC.removeValue(this);
                 parent->savePreference();
                 break;
             case 2:
-                if ( AlertWindow::showYesNoCancelBox(AlertWindow::WarningIcon, "Confirm", "Clear midi mapping for all controller change (CC) messages?", "YES", "NO", "CANCEL") ) {
+                //if ( AlertWindow::showYesNoCancelBox(AlertWindow::WarningIcon, "Confirm", "Clear midi mapping for all controller change (CC) messages?", "YES", "NO", "CANCEL") ) {
                     parent->mappedMidiCC.clear();
                     parent->savePreference();
-                }
+                //}
                 break;
             case 3: {
                 AudioProcessorEditor *editor = parent->getActiveEditor();
@@ -171,6 +172,7 @@ void Ctrl::mouseDown(const juce::MouseEvent &event) {
             }
             break;
         }
+        });
     }
 }
 
diff --git a/Source/PluginProcessor.cpp b/Source/PluginProcessor.cpp
index 369bfd8..ec0630b 100644
--- a/Source/PluginProcessor.cpp
+++ b/Source/PluginProcessor.cpp
@@ -914,27 +914,24 @@ void DexedAudioProcessor::applySCLTuning() {
     File s;
 
     // loop to enforce the proper selection
-    for (;;) {
-        // open file chooser dialog
-        if (!fc.browseForFileToOpen())
-            // User cancelled
-            return;
+    // open file chooser dialog
+    fc.launchAsync(juce::FileBrowserComponent::openMode, [&](const juce::FileChooser &_fc) {
         s = fc.getResult();
 
         // enforce file extenstion ''.scl''.
         // (reason: the extension ''.scl'' is mandatory according to 
         // ''https://www.huygens-fokker.org/scala/scl_format.html''
         if (s.getFileExtension() != ".scl") {
-            AlertWindow::showMessageBox(AlertWindow::WarningIcon, "Invalid file type!", "Only files with the \".scl\" extension (in lowercase!) are allowed.");
-            continue;
+            AlertWindow::showMessageBoxAsync(AlertWindow::WarningIcon, "Invalid file type!", "Only files with the \".scl\" extension (in lowercase!) are allowed.");
+            applySCLTuning();
         }
 
         // enforce to select file below the max limit16KB sized files
         if (s.getSize() > MAX_SCL_KBM_FILE_SIZE) {
             std::string msg;
             msg = "File size exceeded the maximum limit of " + std::to_string(MAX_SCL_KBM_FILE_SIZE) + " bytes.";
-            AlertWindow::showMessageBox(AlertWindow::WarningIcon, "File size error!", msg);
-            continue;
+            AlertWindow::showMessageBoxAsync(AlertWindow::WarningIcon, "File size error!", msg);
+            applySCLTuning();
         }
 
         // enforce to select non-empty file
@@ -942,16 +939,14 @@ void DexedAudioProcessor::applySCLTuning() {
         if (s.getSize() == 0) {
             std::string msg;
             msg = "File is empty.";
-            AlertWindow::showMessageBox(AlertWindow::WarningIcon, "File size error!", msg);
-            continue;
+            AlertWindow::showMessageBoxAsync(AlertWindow::WarningIcon, "File size error!", msg);
+            applySCLTuning();
         }
 
         // try to apply the SCL file 
         applySCLTuning(s);
 
-        // exit the loop
-        break;
-    }
+    });
 }
 
 void DexedAudioProcessor::applySCLTuning(File s) {
@@ -991,26 +986,23 @@ void DexedAudioProcessor::applyKBMMapping() {
     File s;
 
     // loop to enforce the proper selection
-    for (;;) {
-        // invoke file chooser dialog
-        if (!fc.browseForFileToOpen())            
-            return; // User cancelled
+    fc.launchAsync(juce::FileBrowserComponent::openMode, [&](const juce::FileChooser &_fc) {
         s = fc.getResult();
 
         // enforce file extenstion ''.kbm''.
         // (reason: the extension ''.kbm'' is mandatory according to 
         // ''https://www.huygens-fokker.org/scala/scl_format.html''
         if (s.getFileExtension() != ".kbm") {
-            AlertWindow::showMessageBox(AlertWindow::WarningIcon, "Invalid file type!", "Only files with the \".kbm\" extension (in lowercase!) are allowed.");
-            continue;
+            AlertWindow::showMessageBoxAsync(AlertWindow::WarningIcon, "Invalid file type!", "Only files with the \".kbm\" extension (in lowercase!) are allowed.");
+            applyKBMMapping();
         }
 
         // enforce to select file below the max limit16KB sized files
         if (s.getSize() > MAX_SCL_KBM_FILE_SIZE) {
             std::string msg;
             msg = "File size exceeded the maximum limit of " + std::to_string(MAX_SCL_KBM_FILE_SIZE) + " bytes.";
-            AlertWindow::showMessageBox(AlertWindow::WarningIcon, "File size error!", msg);
-            continue;
+            AlertWindow::showMessageBoxAsync(AlertWindow::WarningIcon, "File size error!", msg);
+            applyKBMMapping();
         }
 
         // enforce to select non-empty file
@@ -1018,16 +1010,13 @@ void DexedAudioProcessor::applyKBMMapping() {
         if (s.getSize() == 0) {
             std::string msg;
             msg = "File is empty.";
-            AlertWindow::showMessageBox(AlertWindow::WarningIcon, "File size error!", msg);
-            continue;
+            AlertWindow::showMessageBoxAsync(AlertWindow::WarningIcon, "File size error!", msg);
+            applyKBMMapping();
         }
 
         // try to apply KBM mapping
         applyKBMMapping(s);
-
-        // exit the loop
-        break;
-    }
+    });
 }
 
 void DexedAudioProcessor::applyKBMMapping( File s )
Submodule libs/surgesynthteam_tuningui contains modified content
diff --git a/libs/surgesynthteam_tuningui/surgesynthteam_tuningui.h b/libs/surgesynthteam_tuningui/surgesynthteam_tuningui.h
index fa17052..be7287b 100644
--- a/libs/surgesynthteam_tuningui/surgesynthteam_tuningui.h
+++ b/libs/surgesynthteam_tuningui/surgesynthteam_tuningui.h
@@ -181,15 +181,16 @@ public:
         {
             rmbMenu->clear();
             rmbMenu->addItem(1, "Export to CSV" );
-            auto result = rmbMenu->show();
-            if( result == 1 )
-                exportToCSV();
+            rmbMenu->showMenuAsync({}, [&](int result) {
+                if( result == 1 )
+                    exportToCSV();
+            });
         }
     }
 
     virtual void exportToCSV() {
         juce::FileChooser fc( "Export CSV to...", juce::File(), "*.csv" );
-        if( fc.browseForFileToSave(true) )
+        fc.launchAsync(0, [&](const juce::FileChooser &fc)
         {
             auto f = fc.getResult();
             std::ostringstream csvStream;
@@ -207,6 +208,7 @@ public:
 
             }
         }
+        );
     }
        
 
